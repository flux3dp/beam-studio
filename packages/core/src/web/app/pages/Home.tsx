import * as React from 'react';

import appSettings from 'app/app-settings';
import i18n from 'helpers/i18n';
import menu from 'implementations/menu';
import Modal from 'app/widgets/Modal';
import SelectView from 'app/widgets/Select';

function Home(): JSX.Element {
  const [lang, changeLang] = React.useState(i18n.lang);
  const getLanguageOptions = () => {
    const options = [];
    const langCodes = Object.keys(appSettings.i18n.supported_langs);
    for (let i = 0; i < langCodes.length; i += 1) {
      const langCode = langCodes[i];
      options.push({
        value: langCode,
        label: appSettings.i18n.supported_langs[langCode],
        selected: langCode === i18n.getActiveLang(),
      });
    }
    return options;
  };

  const changeActiveLang = (e) => {
    i18n.setActiveLang(e.currentTarget.value);
    menu.updateLanguage();
    changeLang(i18n.lang);
  };

  const wrapperClassName = {
    initialization: true,
  };
  const content = (
    <div className="home text-center">
      <img className="brand-image" src="img/menu/main_logo.svg" />
      <div>
        <h1 className="headline">{lang.initialize.select_language}</h1>
        <div className="language">
          <SelectView id="select-lang" options={getLanguageOptions()} onChange={changeActiveLang} />
        </div>
        <div>
          <a href="#initialize/connect/flux-id-login" className="btn btn-action btn-large">{lang.initialize.next}</a>
        </div>
      </div>
    </div>
  );
  return (
    <Modal className={wrapperClassName} content={content} />
  );
}

export default Home;
