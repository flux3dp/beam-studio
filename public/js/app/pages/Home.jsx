define([,
    'helpers/i18n',
    'jsx!widgets/Select',
    'jsx!widgets/Modal'
], function(i18n, SelectView, Modal) {
    'use strict';

    const React = require('react');

    return function(args) {
        args = args || {};

        class Home extends React.Component{
            constructor(props) {
                super(props);
                this.state = {
                    lang: args.state.lang
                };
            }

            // Private methods
            _getLanguageOptions = () => {
                var options = [];

                for (var lang_code in args.props.supported_langs) {
                    options.push({
                        value: lang_code,
                        label: args.props.supported_langs[lang_code],
                        selected: lang_code === i18n.getActiveLang()
                    });
                }

                return options;
            }

            _changeActiveLang = (e) => {
                i18n.setActiveLang(e.currentTarget.value);
                this.setState({
                    lang: i18n.get()
                });
            }

            // Lifecycle
            render() {
                var lang = this.state.lang,
                    options = this._getLanguageOptions(),
                    wrapperClassName = {
                        'initialization': true
                    },
                    content = (
                        <div className="home text-center">
                            <img className="brand-image" src="img/menu/main_logo.svg"/>
                            <div>
                                <h1 className="headline">{lang.initialize.select_language}</h1>
                                <div className="language">
                                    <SelectView id="select-lang" options={options} onChange={this._changeActiveLang}/>
                                </div>
                                <div>
                                    <a href="#initialize/wifi/select-machine-type" className="btn btn-action btn-large">{lang.initialize.next}</a>
                                </div>
                            </div>
                        </div>
                    );

                return (
                    <Modal className={wrapperClassName} content={content}/>
                );
            }
        };
        return Home;
    };
});
