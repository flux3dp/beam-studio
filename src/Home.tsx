import * as React from 'react';
import electron from 'electron';

import i18n from 'helpers/i18n';
import Modal from 'app/widgets/Modal';
import SelectView from 'app/widgets/Select';

import communicator from 'implementations/communicator';

const { Menu } = electron.remote;

export default function(args) {
    args = args || {};

    interface State {
      lang: any;
    }

    class Home extends React.Component<any, State> {
        constructor(props) {
            super(props);
            this.state = {
                lang: args.state.lang
            };
        }

        // Private methods
        _getLanguageOptions = () => {
            var options = [];

            for (var lang_code in args.props.supported_langs) {
                options.push({
                    value: lang_code,
                    label: args.props.supported_langs[lang_code],
                    selected: lang_code === i18n.getActiveLang()
                });
            }

            return options;
        }

        _changeActiveLang = (e) => {
            i18n.setActiveLang(e.currentTarget.value);
            communicator.send('NOTIFY_LANGUAGE');
            if (window.os === 'Windows') {
                window['titlebar'].updateMenu(Menu.getApplicationMenu());
            }
            this.setState({
                lang: i18n.lang
            });
        }

        // Lifecycle
        render() {
            var lang = this.state.lang,
                options = this._getLanguageOptions(),
                wrapperClassName = {
                    'initialization': true
                },
                content = (
                    <div className="home text-center">
                        <img className="brand-image" src="img/menu/main_logo.svg"/>
                        <div>
                            <h1 className="headline">{lang.initialize.select_language}</h1>
                            <div className="language">
                                <SelectView id="select-lang" options={options} onChange={this._changeActiveLang}/>
                            </div>
                            <div>
                                <a href="#initialize/connect/flux-id-login" className="btn btn-action btn-large">{lang.initialize.next}</a>
                            </div>
                        </div>
                    </div>
                );

            return (
                <Modal className={wrapperClassName} content={content}/>
            );
        }
    };
    return Home;
};
