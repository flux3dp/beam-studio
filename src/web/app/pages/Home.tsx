import i18n from '../../helpers/i18n';
import SelectView from '../widgets/Select';
import Modal from '../widgets/Modal';

const React = requireNode('react');
const electron = requireNode('electron');
const { Menu } = electron.remote;

export default function(args) {
    args = args || {};

    class Home extends React.Component{
        constructor(props) {
            super(props);
            this.state = {
                lang: args.state.lang
            };
        }

        // Private methods
        _getLanguageOptions = () => {
            var options = [];

            for (var lang_code in args.props.supported_langs) {
                options.push({
                    value: lang_code,
                    label: args.props.supported_langs[lang_code],
                    selected: lang_code === i18n.getActiveLang()
                });
            }

            return options;
        }

        _changeActiveLang = (e) => {
            i18n.setActiveLang(e.currentTarget.value);
            electron.ipcRenderer.send('NOTIFY_LANGUAGE');
            if (process.platform === 'win32') {
                window['titlebar'].updateMenu(Menu.getApplicationMenu());
            }
            this.setState({
                lang: i18n.lang
            });
        }

        // Lifecycle
        render() {
            var lang = this.state.lang,
                options = this._getLanguageOptions(),
                wrapperClassName = {
                    'initialization': true
                },
                content = (
                    <div className="home text-center">
                        <img className="brand-image" src="img/menu/main_logo.svg"/>
                        <div>
                            <h1 className="headline">{lang.initialize.select_language}</h1>
                            <div className="language">
                                <SelectView id="select-lang" options={options} onChange={this._changeActiveLang}/>
                            </div>
                            <div>
                                <a href="#initialize/connect/flux-id-login" className="btn btn-action btn-large">{lang.initialize.next}</a>
                            </div>
                        </div>
                    </div>
                );

            return (
                <Modal className={wrapperClassName} content={content}/>
            );
        }
    };
    return Home;
};
