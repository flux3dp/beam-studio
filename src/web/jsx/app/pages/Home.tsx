function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

define(['helpers/i18n', 'jsx!widgets/Select', 'jsx!widgets/Modal'], function (i18n, SelectView, Modal) {
  'use strict';

  const React = require('react');

  return function (args) {
    args = args || {};

    class Home extends React.Component {
      constructor(props) {
        super(props);

        _defineProperty(this, "_getLanguageOptions", () => {
          var options = [];

          for (var lang_code in args.props.supported_langs) {
            options.push({
              value: lang_code,
              label: args.props.supported_langs[lang_code],
              selected: lang_code === i18n.getActiveLang()
            });
          }

          return options;
        });

        _defineProperty(this, "_changeActiveLang", e => {
          i18n.setActiveLang(e.currentTarget.value);
          this.setState({
            lang: i18n.get()
          });
        });

        this.state = {
          lang: args.state.lang
        };
      } // Private methods


      // Lifecycle
      render() {
        var lang = this.state.lang,
            options = this._getLanguageOptions(),
            wrapperClassName = {
          'initialization': true
        },
            content = /*#__PURE__*/React.createElement("div", {
          className: "home text-center"
        }, /*#__PURE__*/React.createElement("img", {
          className: "brand-image",
          src: "img/menu/main_logo.svg"
        }), /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("h1", {
          className: "headline"
        }, lang.initialize.select_language), /*#__PURE__*/React.createElement("div", {
          className: "language"
        }, /*#__PURE__*/React.createElement(SelectView, {
          id: "select-lang",
          options: options,
          onChange: this._changeActiveLang
        })), /*#__PURE__*/React.createElement("div", null, /*#__PURE__*/React.createElement("a", {
          href: "#initialize/connect/select-connection-type",
          className: "btn btn-action btn-large"
        }, lang.initialize.next))));

        return /*#__PURE__*/React.createElement(Modal, {
          className: wrapperClassName,
          content: content
        });
      }

    }

    ;
    return Home;
  };
});