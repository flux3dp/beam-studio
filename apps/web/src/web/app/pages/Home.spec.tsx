import classNames from 'classnames';
import React from 'react';
import { render } from '@testing-library/react';

jest.mock('helpers/i18n', () => ({
  getActiveLang: () => 'en',
  setActiveLang: jest.fn(),
  lang: {
    initialize: {
      select_language: 'Select Language',
      next: 'Next',
    },
  },
}));

jest.mock('implementations/menu', () => ({
  updateLanguage: jest.fn(),
}));

jest.mock('app/widgets/Modal', () => ({ className, content }: any) => (
  <div className={classNames(className)}>{content}</div>
));

// eslint-disable-next-line import/first
import Home from './Home';

test('should render correctly', () => {
  const { container } = render(<Home />);
  expect(container).toMatchSnapshot();
});
