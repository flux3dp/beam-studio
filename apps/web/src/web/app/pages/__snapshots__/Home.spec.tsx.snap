// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <div
    class="initialization"
  >
    <div
      class="home text-center"
    >
      <img
        class="brand-image"
        src="img/menu/main_logo.svg"
      />
      <div>
        <h1
          class="headline"
        >
          Select Language
        </h1>
        <div
          class="language"
        >
          <select
            id="select-lang"
          >
            <option
              label="Czech"
              value="cs"
            >
              Czech
            </option>
            <option
              label="Dansk"
              value="da"
            >
              Dansk
            </option>
            <option
              label="Deutsch"
              value="de"
            >
              Deutsch
            </option>
            <option
              label="Ελληνικά"
              value="el"
            >
              Ελληνικά
            </option>
            <option
              label="English"
              value="en"
            >
              English
            </option>
            <option
              label="Español"
              value="es"
            >
              Español
            </option>
            <option
              label="Suomi"
              value="fi"
            >
              Suomi
            </option>
            <option
              label="Français"
              value="fr"
            >
              Français
            </option>
            <option
              label="Bahasa Indonesia"
              value="id"
            >
              Bahasa Indonesia
            </option>
            <option
              label="Italiano"
              value="it"
            >
              Italiano
            </option>
            <option
              label="日本語"
              value="ja"
            >
              日本語
            </option>
            <option
              label="한국어"
              value="kr"
            >
              한국어
            </option>
            <option
              label="Melayu"
              value="ms"
            >
              Melayu
            </option>
            <option
              label="Nederlands"
              value="nl"
            >
              Nederlands
            </option>
            <option
              label="Norsk"
              value="no"
            >
              Norsk
            </option>
            <option
              label="Polski"
              value="pl"
            >
              Polski
            </option>
            <option
              label="Português"
              value="pt"
            >
              Português
            </option>
            <option
              label="Svenska"
              value="se"
            >
              Svenska
            </option>
            <option
              label="ภาษาไทย"
              value="th"
            >
              ภาษาไทย
            </option>
            <option
              label="Tiếng Việt"
              value="vi"
            >
              Tiếng Việt
            </option>
            <option
              label="简体中文"
              value="zh-cn"
            >
              简体中文
            </option>
            <option
              label="繁體中文"
              value="zh-tw"
            >
              繁體中文
            </option>
          </select>
        </div>
        <div>
          <a
            class="btn btn-action btn-large"
            href="#initialize/connect/flux-id-login"
          >
            Next
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
